### PR TITLE
Create UI states

### PR DESCRIPTION
**Pull request by andresmr**

### CODE REVIEW

**Review by mvmmacamo2 — CHANGES_REQUESTED** [review]

File: stock-managment/src/main/java/org/dhis2/android/rtsm/ui/home/HomeViewModel.kt [hunk 1]
```diff
@@ -42,39 +44,14 @@ class HomeViewModel @Inject constructor(
...
-
     private val _scanText = MutableStateFlow("")
     val scanText = _scanText.asStateFlow()
 
-    private val _oldSelectedFacility = MutableStateFlow("")
-    val oldSelectedFacility = _oldSelectedFacility.asStateFlow()
-
     private val _transactionDate = MutableStateFlow(LocalDateTime.now())
     val transactionDate: StateFlow<LocalDateTime>
         get() = _transactionDate
```
[the remark this answers is not in the document]
Comment (andresmr) [reply stock-managment/src/main/java/org/dhis2/android/rtsm/ui/home/HomeViewModel.kt:52]: you are right, good feedback!. I'll do it.

**Review by CarlosMacaneta — APPROVED** [review]

File: stock-managment/src/main/java/org/dhis2/android/rtsm/ui/home/screens/components/MainContent.kt [hunk 2]
```diff
@@ -209,30 +209,30 @@ fun MainContent(
...
                         scope.launch { backdropState.conceal() }
                     }
                 }
-            } else if (viewModel.toolbarTitle.collectAsState().value.name
-                == TransactionType.DISCARD.name
+            } else if (viewModel.settingsUiState.collectAsState().value.transactionType
+                == TransactionType.DISCARD
             ) {
-                if (viewModel.hasFacilitySelected.collectAsState().value) {
+                if (hasFacilitySelected) {
```
[the remark this answers is not in the document]
Comment (andresmr) [reply stock-managment/src/main/java/org/dhis2/android/rtsm/ui/home/screens/components/MainContent.kt:235]: You are right, changing it...

File: stock-managment/src/main/java/org/dhis2/android/rtsm/ui/home/HomeViewModel.kt [hunk 3]
```diff
@@ -91,8 +61,6 @@ class HomeViewModel @Inject constructor(
         get() = _destinations
 
     // Toolbar section variables
-    private val _toolbarTitle = MutableStateFlow(TransactionType.DISTRIBUTION)
-    val toolbarTitle: StateFlow<TransactionType> get() = _toolbarTitle
 
     private val _fromFacility = MutableStateFlow(UIText.StringRes(R.string.from_facility))
```
Comment (CarlosMacaneta) [inline stock-managment/src/main/java/org/dhis2/android/rtsm/ui/home/HomeViewModel.kt:65]: Why don't we move this to settingsuistate or create a toolbaruistate to listen this changes?

**Review by mvmmacamo2 — APPROVED** [review]

File: stock-managment/src/main/java/org/dhis2/android/rtsm/ui/home/screens/components/DropdownComponent.kt [hunk 4]
```diff
@@ -457,7 +459,8 @@ fun DropdownComponentDistributedTo(
 fun openOrgUnitTreeSelector(
     supportFragmentManager: FragmentManager,
     data: List<OrganisationUnit>,
-    viewModel: HomeViewModel
+    viewModel: HomeViewModel,
+    settingsUiState: SettingsUiState
 ) {
     val programUid = "F5ijs28K4s8"
```
Comment (CarlosMacaneta) [inline stock-managment/src/main/java/org/dhis2/android/rtsm/ui/home/screens/components/DropdownComponent.kt:465]: why are we using hardcoded program uid?

**Review by CarlosMacaneta — CHANGES_REQUESTED** [review]

**Review by CarlosMacaneta — APPROVED** [review]

**Pull request merged**